src/Network/CurlResult.php

```diff
@@ -226,11 +226,62 @@ public function getContentType()
 	/**
 	 * Returns the Curl headers
 	 *
-	 * @return string the Curl headers
+	 * @param string $field optional header field. Return all fields if empty
+	 *
+	 * @return string the Curl headers or the specified content of the header variable
+	 */
+	public function getHeader(string $field = '')
+	{
+		if (empty($field)) {
+			return $this->header;
+		}
+
+		$field = strtolower(trim($field));
+
+		$headers = self::getHeaderArray();
+
+		if (isset($headers[$field])) {
+			return $headers[$field];
+		}
+	}
+
+	/**
+	 * Check if a specified header exists
+	 *
+	 * @param string $field header field
+	 *
+	 * @return boolean "true" if header exists
 	 */
-	public function getHeader()
+	public function headerExists(string $field)
 	{
-		return $this->header;
+		$field = strtolower(trim($field));
+
+		$headers = self::getHeaderArray();
+
+		return array_key_exists($field, $headers);
+	}
+
+	/**
+	 * Returns the Curl headers as an associated array
+	 *
+	 * @return array associated header array
+	 */
+	public function getHeaderArray()
```

Comment:
Please perform this once in `parseBodyHeader` and store the result in a class variable.

Reply:
For performance reasons I don't want to perform this for every curl execution, since at most requests you don't look at the headers.

Reply:
But when you do look at the headers, you will perform this at least twice for each expected header.

Reply:
Have a look at the changed code. I implemented some caching.

Reply:
Great, thanks!